src/System.Data.SqlClient/src/System/Data/SqlClient/SqlClientDiagnosticListenerExtensions.cs

```diff
@@ -205,5 +217,182 @@ public static void WriteConnectionCloseError(this DiagnosticListener @this, Guid
                     });
             }
         }
+
+        public static Guid WriteTransactionCommitBefore(this DiagnosticListener @this, IsolationLevel isolationLevel, SqlConnection connection, [CallerMemberName] string operation = "")
+        {
+            if (@this.IsEnabled(SqlBeforeCommitTransaction))
+            {
+                Guid operationId = Guid.NewGuid();
+
+                @this.Write(
+                    SqlBeforeCommitTransaction,
+                    new
+                    {
+                        OperationId = operationId,
+                        Operation = operation,
+                        IsolationLevel = isolationLevel,
+                        Connection = connection,
+                        Timestamp = Stopwatch.GetTimestamp()
+                    });
+
+                return operationId;
+            }
+            else
+                return Guid.Empty;
+        }
+
+        public static void WriteTransactionCommitAfter(this DiagnosticListener @this, Guid operationId, IsolationLevel isolationLevel, SqlConnection connection, [CallerMemberName] string operation = "")
```

Comment:
Instead of `WriteTransactionCommitAfter`  `WriteAfterTransactionCommit` would be better